core/src/main/scala/it/scalachess/core/board/Position.scala

```diff
@@ -20,6 +20,9 @@ final case class Position(col: Int, row: Int) {
   lazy val upLeft: Option[Position]    = Position.of(col - 1)(row + 1)
   lazy val upRight: Option[Position]   = Position.of(col + 1)(row + 1)
 
+  lazy val adjacentPositions: Set[Option[Position]] =
+    Set(left, right, down, up, downLeft, downRight, upLeft, upRight)
+
   /**
```

Comment:
No need for Option[Position]
` lazy val adjacentPositions: Set[Position] =
    Set(left, right, down, up, downLeft, downRight, upLeft, upRight).flatten`